bindings_ffi/src/mls.rs

```diff
@@ -1047,15 +1057,18 @@ mod tests {
 
     async fn new_test_client() -> Arc<FfiXmtpClient> {
         let ffi_inbox_owner = LocalWalletInboxOwner::new();
+        let nonce = 1;
```

Comment:
Why a nonce of 1 here?

Reply:
It doesn't provide a legacy key so it cannot use nonce 0.

Reply:
Hmm. @richardhuaaa what do you think of that change to the protocol? In the XIP we don’t make any restrictions on using nonce 0 without a legacy key. 

But it does preserve nonce 0 for later use with a legacy key if we did.

Reply:
Why is nonce 0 preserved for legacy keys?

Currently, if a wallet has registered with an inbox, next time it won't be able to register another inbox with nonce 0 & legacy key unless it revoked the first inbox.(I know @neekolas mentioned that it doesn't need to revoke the first inbox but only needs to register with the new inbox but I'm not sure if that works. Because the installation keys added by the wallet in the first inbox will not be removed if the wallet is not specifically revoked in the first inbox)

Would love the clarity on this - want to make sure the identity creation logic is solid

Reply:
Correct me if I'm wrong - this `only be used once` has yet to be implemented in libxmtp
```
XIP-46 

// This signature type can ONLY be used for CreateInbox and AddAssociation
// payloads, and can only be used once in xmtpv3.
message LegacyDelegatedSignature {
  xmtp.message_contents.SignedPublicKey delegated_key = 1;
  RecoverableEcdsaSignature signature = 2;
}

```

Reply:
> Why is nonce 0 preserved for legacy keys?

For context, this is what the XIP states:
> The ‘Create Identity’ signature of the delegated_key is implicitly treated as a CreateInbox action, with a nonce of 0.

So we use nonce 0 here because we don't have any way to retroactively specify a nonce for a `Create Identity` signature from v2.

> Currently, if a wallet has registered with an inbox, next time it won't be able to register another inbox with nonce 0 & legacy key unless it revoked the first inbox.

Is this purely an issue for unit tests, where we might want to create multiple Inbox ID's for test purposes? In a production environment, we want a wallet (and all of its installations) to always resolve to the same inbox, so there's no conflict if a legacy key registers the inbox ID with nonce 0, and a later installation uses the same nonce (or vice versa).

Reply:
> `this only be used once` has yet to be implemented in libxmtp

It's enforced in the identity update processing. Legacy signatures have to be used on nonce 0, and it will trigger a replay error if it is used more than once in a sequence of identity updates.

Reply:
> the installation keys added by the wallet in the first inbox will not be removed if the wallet is not explicitly revoked in the first inbox

For all cases where you are moving inbox_ids we can assume that you'll be working with a fresh DB and new installation keys